exist-core/src/main/java/org/exist/util/Configuration.java

```diff
@@ -898,722 +950,629 @@ private void configureBackend( final Optional<Path> dbHome, Element con ) throws
-    		} else {
-    			value = element.getAttribute( attributeName );
-    		}
-    	}
-    	
-    	return( value );
+    private String getConfigAttributeValue(final Element element, final String attributeName) {
+        if (element == null || attributeName == null) {
+            return null;
+        }
+
+        final String property = getAttributeSystemPropertyName(element, attributeName);
+        final String value = System.getProperty(property);
+
+        if (value != null) {
+            LOG.warn("Configuration value overridden by system property: {}, with value: {}", property, value);
+            return value;
+        }
+        // If the value has not been overridden in a system property, then get it from the configuration
+        return element.getAttribute(attributeName);
     }
-    
+
     /**
      * Generates a suitable system property name from the given config attribute and parent element.
-     *
-     * values are of the form org.element.element.....attribute and follow the heirarchical structure of the conf.xml file. 
+     * <p>
+     * values are of the form org.element.element.....attribute and follow the hierarchical structure of the conf.xml file.
      * For example, the db-connection cacheSize property name would be org.exist.db-connection.cacheSize
      *
-     * @param   element        The attribute's parent element
-     * @param   attributeName  The name of the attribute
-     *
-     * @return  The generated system property name
+     * @param element       The attribute's parent element
+     * @param attributeName The name of the attribute
+     * @return The generated system property name
      */
-    private String getAttributeSystemPropertyName( Element element, String attributeName )
-    {
-    	final StringBuilder  	property 	= new StringBuilder( attributeName );
-    	Node			parent		= element.getParentNode();
-    	
-    	property.insert( 0, "." );
-    	property.insert( 0,  element.getLocalName() );
-    	
-    	while( parent != null && parent instanceof Element ) {
-    		final String parentName = ((Element)parent).getLocalName();
-    		
-    		property.insert( 0, "." );
-    		property.insert( 0, parentName );
-    		
-    		parent   = parent.getParentNode();
-    	}
-    	
-    	property.insert( 0, "org." );
-    	
-    	return( property.toString() );
-    }
+    private String getAttributeSystemPropertyName(Element element, String attributeName) {
+        final StringBuilder property = new StringBuilder(attributeName);
+        Node parent = element.getParentNode();
+
+        property.insert(0, ".");
+        property.insert(0, element.getLocalName());
+
+        while (parent instanceof Element) {
+            final String parentName = parent.getLocalName();
+
+            property.insert(0, ".");
+            property.insert(0, parentName);
 
+            parent = parent.getParentNode();
+        }
+
+        property.insert(0, "org.");
+
+        return (property.toString());
```

Comment:
The surrounding parentheses can be removed I guess..

Reply:
I believe so, too.

Reply:
@line-o could you remove the parentheses in order I can merge?